pkg/monitor/worker.go

```diff
@@ -255,5 +302,24 @@ func (mon *monitor) workOne(ctx context.Context, log *logrus.Entry, doc *api.Ope
 		return
 	}
 
-	c.Monitor(ctx)
+	monitors = append(monitors, c)
+	//c.Monitor(ctx)
+	allJobsDone := make(chan bool)
+	go execute(ctx, allJobsDone, wg, monitors)
+
+	select {
+	case <-allJobsDone:
+	case <-ctx.Done():
+		log.Infof("NSG Monitoring processing for cluster %s has timed out", doc.OpenShiftCluster.ID)
```

Comment:
this could be related to the normal monitor failing as well, correct?

Reply:
made this into a general message.